android/src/main/java/io/invertase/firebase/notifications/RNFirebaseBackgroundNotificationReceiver.java

```diff
@@ -0,0 +1,47 @@
+package io.invertase.firebase.notifications;
+
+import android.app.ActivityManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import com.facebook.react.HeadlessJsTaskService;
+
+import java.util.List;
+
+public class RNFirebaseBackgroundNotificationReceiver extends BroadcastReceiver {
+
+  @Override
+  public void onReceive(Context context, Intent intent) {
+    Intent serviceIntent = new Intent(context, RNFirebaseBackgroundNotificationsService.class);
+    serviceIntent.putExtras(intent.getExtras());
+    context.startService(serviceIntent);
+
+    if (!isAppOnForeground((context))) {
```

Comment:
What happens if the app is in the foreground when the `Snooze` action is hit?
I _think_ it will currently just get lost - instead we need to make sure that the action gets passed through into the standard `onNotificationOpened` handler so that it can be handled normally.

Reply:
I need to test this. I'm not sure if going through standard `onNotificationOpened` way is the best solution, since this would require code _duplication_ (the same handler function should be called from two places). Would rather handle everything in background.

There is a small drawback for the _always background_ approach. When background action handler is changing the UI state this should be achieved using some kind of event or listener approach.

Reply:
I believe it would be better to handle things separately in the foreground and background as it gives more flexibility to the developer if they're like their app to act differently.  You've already touched on UI state which might need to be updated in the foreground.

The functionality that's shared can easily be wrapped inside a method which can be called from both the background and foreground handler to prevent code duplication.